apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -560,3 +554,104 @@ function wpcom_is_previewing_global_styles( ?int $user_id = null ) {
 	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
 	return ! isset( $_GET['hide-global-styles'] ) && user_can( $user_id, 'administrator' );
 }
+
+/**
+ * Checks whether the site has access to Global Styles with a Personal plan as part of an A/B test.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether the site has access to Global Styles with a Personal plan.
+ */
+function wpcom_site_has_global_styles_in_personal_plan( $blog_id = 0 ) {
+	if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+		return false;
+	}
+
+	if ( ! $blog_id ) {
+		$blog_id = get_current_blog_id();
+	}
+
+	$cache_key             = "global-styles-on-personal-assignment-$blog_id";
+	$experiment_assignment = wp_cache_get( $cache_key, 'a8c_experiments', false, $found_in_cache );
+	if ( $found_in_cache ) {
+		return 'treatment' === $experiment_assignment;
+	}
+
+	$owner_id = wpcom_get_blog_owner( $blog_id );
+	if ( ! $owner_id ) {
+		return false;
+	}
+
+	$owner = get_userdata( $owner_id );
+	if ( ! $owner ) {
+		return false;
+	}
+
+	$experiment_assignment = \ExPlat\get_user_assignment( 'calypso_global_styles_personal', $owner );
+	// Cache the experiment assignment to prevent duplicate DB queries in the frontend.
+	wp_cache_set( $cache_key, $experiment_assignment, 'a8c_experiments', MONTH_IN_SECONDS );
+	return 'treatment' === $experiment_assignment;
+}
+
+/**
+ * Checks whether the site has a Personal plan.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether the site has a Personal plan.
+ */
+function wpcom_site_has_personal_plan( $blog_id ) {
+	$personal_plans = array_filter(
+		wpcom_get_site_purchases( $blog_id ),
+		function ( $purchase ) {
+			return strpos( $purchase->product_slug, 'personal-bundle' ) === 0;
+		}
+	);
+
+	return ! empty( $personal_plans );
+}
+
+/**
+ * Checks whether the site has a plan that grants access to the Global Styles feature.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether the site has access to Global Styles.
+ */
+function wpcom_site_has_global_styles_feature( $blog_id = 0 ) {
+	/*
+	 * Non-Simple sites on a lower plan are temporary edge cases. We grant them access
+	 * to Global Styles to prevent unexpected temporary changes in their styles.
+	 */
+	if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+		return true;
+	}
+
+	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
+		return true;
+	}
+
+	/*
+	 * Users who were assigned to the treatment group of the Global Styles on Personal experiment
+	 * while having a Personal plan should always have access to Global Styles, as long as they still
+	 * have a Personal plan.
+	 */
+	if ( wpcom_global_styles_has_blog_sticker( 'wpcom-global-styles-personal-plan', $blog_id ) ) {
+		return wpcom_site_has_personal_plan( $blog_id );
+	}
```

Comment:
With this code, once a site has the `wpcom_global_styles_has_blog_sticker` when will line `651` get executed?

Reply:
> once a site has the `wpcom_global_styles_has_blog_sticker` sticker

I think you mean once a site has the `wpcom-global-styles-personal-plan` sticker, right? Line `651` will never be reached for those sites.

Reply:
https://github.com/Automattic/wp-calypso/pull/78771/commits/9daca8bdad135029f7844ed91e4a0169d11e04c2

Reply:
Yes, it's `wpcom-global-styles-personal-plan` sorry for the confusion. Nice, now the sticker can get removed!

Reply:
Nope, we need it in case that `control` wins. See paYJgx-3yE-p2